core/src/main/scala/org/apache/spark/ui/jobs/PoolPage.scala

```diff
@@ -31,7 +31,7 @@ private[ui] class PoolPage(parent: StagesTab) extends WebUIPage("pool") {
 
   def render(request: HttpServletRequest): Seq[Node] = {
     listener.synchronized {
-      val poolName = request.getParameter("poolname")
+      val poolName = UIUtils.decodeURLParameter(request.getParameter("poolname"))
```

Comment:
NullPointerException if `poolname` is not set. Could you use `Option....getOrElse` like other place?